starfish/test/full_pipelines/cli/test_help.py

```diff
@@ -0,0 +1,120 @@
+import difflib
+import subprocess
+import unittest
+
+from starfish.util import exec
+
+
+def assert_diff(actual, expected):
+    if isinstance(actual, bytes):
+        actual = actual.decode("utf-8")
+    if actual != expected:
+        diff = difflib.ndiff(expected.splitlines(keepends=True),
+                             actual.splitlines(keepends=True))
+        diff = list(diff)
+        for line in diff:
+            print(line, end="")
+        raise Exception("help mismatch!")
+
+
+class TestHelpReturnCode(unittest.TestCase):
+    """
+    Tests that the CLI supports a '--help' option at all of the expected levels.
+    """
+
+    STAGES = (
+        [
+            "starfish", "--help",
+            lambda tempdir: tempdir
+        ],
+        [
+            "starfish", "detect_spots", "--help",
+            lambda tempdir: tempdir
+        ],
+        [
+            "starfish", "detect_spots", "BlobDetector", "--help",
+            lambda tempdir: tempdir
+        ],
+    )
+
+    def test_run_build(self):
+        exec.stages(
+            TestHelpReturnCode.STAGES,
+            keep_data=False)
+
+class TestHelpStandardOut(unittest.TestCase):
+    """
+    Tests that the calls to CLI's help produce the output that users expect.
+    """
+
+    def test_first(self):
+        actual = subprocess.check_output(["starfish", "--help"])
+        expected = """Usage: starfish [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  --profile
+  --help     Show this message and exit.
+
+Commands:
+  build
+  decode
+  detect_spots
+  filter
+  registration
+  segment
+  target_assignment
+  validate           invokes validate with the parsed commandline...
+  version
+"""
+        assert_diff(actual, expected)
+
+    def test_second(self):
```

Comment:
These tests seem like they might be a bit annoying to refactor. Would it be adequate to code up something like the following for the purposes of these tests? 
```python3
help_string = "  --help                          Show this message and exit."
assert help_string in actual
```

Reply:
You're right that this is pretty brittle. For example, the test will need to be updated for gh-904 already. I defer on to what degree we want the help output to count as public API which can't break. If we want to go for a less-strict solution, I'd imagine the `Usage:` line would be best, since that is what really broke with this PR, `starfish a b --help` was producing `starfish a --help b` output.

Reply:
Pushed the simplify tests. Waiting on green.